src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -487,7 +489,9 @@ private bool CheckTypesCore(CheckTypesContext context, TexlNode[] args, DType[]
                 }
 
                 var type = argTypes[i];
-                if (type.IsError)
+                if (type.IsError || 
+                    (type.IsVoid && 
+                    !(Name == BuiltinFunctionsCore.With.Name)))
```

Comment:
The base class should not know about the derived class. This should probably be handled in the With function itself, saying that it accepts Void values (in its second argument, not the first)

Reply:
updated!